iXBRLViewerPlugin/iXBRLViewer.py

```diff
@@ -309,21 +306,28 @@ def __init__(self, dts):
     def addFile(self, ivf):
         self.files.append(ivf)
 
-    def save(self, outPath):
+    def save(self, out, outDir=None, responseZip=None):
         """
         Save the iXBRL viewer
         """
-        if os.path.isdir(outPath):
-            # If output is a directory, write each file in the doc set to that
+        if responseZip or os.path.isdir(out):
+            # If output is a zip or directory, write each file in the doc set to that
             # directory using its existing filename
             for f in self.files:
-                filename = os.path.join(outPath, f.filename)
-                self.dts.info("viewer:info", "Writing %s" % filename)
-                with open(filename, "wb") as fout:
-                    writer = XHTMLSerializer()
-                    writer.serialize(f.xmlDocument, fout)
+                self.dts.info("viewer:info", "Writing %s" % f.filename)
+                writer = XHTMLSerializer()
+                xhtmlStr = writer.serialize(f.xmlDocument)
+                if responseZip:
+                    responseZip.writestr(f.filename, xhtmlStr)
```

Comment:
I think this is the key change that's actually needed in this file, although it needs rewriting to use the new signature for writer.serialize which takes a stream (probably needs io.BytesIO)